src/activity_handlers.cpp

```diff
@@ -1274,6 +1275,38 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
     resume_for_multi_activities( *p );
 }
 
+void activity_handlers::milk_finish( player_activity *act, player *p )
+{
+    if( act->coords.empty() ) {
+        debugmsg( "milking activity with no position of monster stored" );
+        return;
+    }
+    const tripoint source_pos = g->m.getlocal( act->coords.at( 0 ) );
+    Creature *c = g->critter_at( source_pos );
```

Comment:
```suggestion
    monster *source_mon = g->critter_at<monster>( source_pos );
```
So you don't have to cast it yourself.